src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,116 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size;
+    private Node<K, V>[] values;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        values = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size >= capacity * LOAD_FACTOR) {
+            resize();
+        }
+        int indexByTheKey = getIndexByTheKey(key);
+        Node<K, V> node = values[indexByTheKey];
+        if (node == null) {
+            values[indexByTheKey] = new Node<>(key, value);
+            size++;
+        } else {
+            while (node != null) {
+                if ((node.key != null) && (node.key.equals(key))
+                        || (node.key == null && key == null)) {
+                    node.value = value;
+                    return;
+                }
+                if (node.next == null) {
+                    node.next = new Node<>(key, value);
+                    size++;
+                    return;
+                }
+                node = node.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int indexByTheKey = getIndexByTheKey(key);
+        Node<K, V> node = values[indexByTheKey];
+        while (node != null) {
+            if ((key == null && node.key == null) || (key != null && key.equals(node.key))) {
```

Comment:
key == null && node.key == null We can simplify this entry by writing one comparison instead of two